src/modules/plus/filter_spot_remover.c

```diff
@@ -0,0 +1,232 @@
+ * Copyright (c) 2018 Meltytech, LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt.h>
+#include <framework/mlt_log.h>
+
+#include <math.h>
+#include <string.h>
+
+/** Scale a rectangle by the specified factors. */
+mlt_rect scale_rect( mlt_rect rect, double x_scale, double y_scale )
+{
+	rect.x = rect.x / x_scale;
+	rect.y = rect.y / y_scale;
+	rect.w = rect.w / x_scale;
+	rect.h = rect.h / y_scale;
+	return rect;
+}
+
+/** Constrain a rect to be within the max dimensions with an additional 1 pixel
+  * padding.
+  */
+mlt_rect constrain_rect( mlt_rect rect, int max_x, int max_y )
+{
+	rect.x = round( rect.x );
+	rect.y = round( rect.y );
+	rect.w = round( rect.w );
+	rect.h = round( rect.h );
+	if ( rect.x < 0 )
+	{
+		rect.w = rect.w + rect.x - 1;
+		rect.x = 1;
+	}
+	if ( rect.y < 0 )
+	{
+		rect.h = rect.h + rect.y - 1;
+		rect.y = 1;
+	}
+	if ( rect.x + rect.w < 0 )
+	{
+		rect.w = 0;
+	}
+	if ( rect.y + rect.h < 0 )
+	{
+		rect.h = 0;
+	}
+	if ( rect.x < 1 )
+	{
+		rect.x = 1;
+	}
+	if ( rect.y < 1 )
+	{
+		rect.y = 1;
+	}
+	if ( rect.x + rect.w > max_x - 1 )
+	{
+		rect.w = max_x - rect.x - 1;
+	}
+	if ( rect.y + rect.h > max_y - 1 )
+	{
+		rect.h = max_y - rect.y - 1;
+	}
+	return rect;
+}
+
+/** Perform spot removal on a channel.
+  *
+  * Values within the rectangle are replaced with interpolated values.
+  * Each value is an interpolation of the first values outside of the rect on
+  * the top, bottom, left and right of the value being interpolated.
+  *
+  * \param chan a pointer to the first value in the channel
+  * \param rowCount the number of values in each line (row)
+  * \param step the space between values in each line
+  * \param rect the area to be removed
+  */
+static void remove_spot_channel( uint8_t *chan, int rowCount, int step, mlt_rect rect )
+{
+	int yStop = rect.y + rect.h;
+	int xStop = rect.x + rect.w;
+	int rowSize = rowCount * step;
+	for ( int y = rect.y; y < yStop; y++ )
+	{
+		uint8_t* xValueL = chan + ( y * rowSize ) + ( ( (int)rect.x - 1 ) * step );
+		uint8_t* xValueR = xValueL + ( (int)rect.w * step );
+		uint8_t* p = chan + ( y * rowSize ) + ( (int)rect.x * step );
+		float yRatio = 1.0 - ( ( y - rect.y ) / rect.h );
```

Comment:
This is a nitpick, but I would have chosen double here especially since the literal value `1.0` is double. Add 'f' after a literal numeric to make it float.

Reply:
Good point. Fixed it here https://github.com/mltframework/mlt/commit/8dd6f12d7dd543109288982d0a8a0b820d620c54 along with another typo: